composer/core/engine.py

```diff
@@ -293,6 +294,10 @@ def run_event(
             self._run_loggers(event)
             self._run_nonlogger_callbacks(event)
             traces = self._run_algorithms(event)
+
+            # If a MosaicMLLogger is present, log analytics for the run to metadata.
+            log_run_analytics(self.logger.destinations)
+
```

Comment:
`LoggerDestination`s are callbacks, which means they run on events. Instead of directly calling in engine.py, you should add `def init` (see other callbacks) to MosaicMLLogger and call your helper function there

Reply:
I think putting the `log_run_analytics` call here would make it run at every `Event` instead of at every step -- is that the intended behavior? As Mihir said, having this run at specific `Event`s is a better way to go about it -- see [here](https://docs.mosaicml.com/projects/composer/en/stable/trainer/callbacks.html#:~:text=Callbacks%20provide%20hooks%20that%20run,%2C%20timing%2C%20or%20model%20introspection.) for more details